global_helpers/panther_oss_helpers.py

```diff
@@ -434,7 +434,7 @@ def listify(maybe_list):
         # not a list
         return [maybe_list]
     # either a list or string
-    return maybe_list if not isinstance(maybe_list, (str, bytes)) else [maybe_list]
+    return maybe_list if not isinstance(maybe_list, (str, bytes, dict)) else [maybe_list]
```

Comment:
IMO we should be adding test cases for this to start contracting the behavior. I also think we should peep through current callers and see what's likely to :boom: 

I read the current return as 
```python
if isinstance(maybe_list, (str, bytes, dict)):
   return [maybelist]
else
   return maybelist
```

Reply:
uh... yeah... this function is weird. maybe I read the return backwards? 

```shell
>>> herps = "a string thing"
>>> iter(herps)
<str_iterator object at 0x104a85360>
>>> herps = b"bytearray"
>>> iter(herps)
<bytes_iterator object at 0x104a85c30>
>>> type(herps)
<class 'bytes'>
>>> herps = {"a dict": 1}
>>> iter(herps)
<dict_keyiterator object at 0x1049b6430>
>>> herps = datetime.datetime.utcnow()
>>> iter(herps)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'datetime.datetime' object is not iterable
```

Reply:
yeah you're right. let me add a test case to the detection where this bug was first spotted to document that this change fixes it.

Reply:
> uh... yeah... this function is weird. maybe I read the return backwards?

the check for iter threw me off a bit too, not sure why it's needed, but i'd hesitate to remove it. and yeah the return is the other way i think: if it's a str,bytes, or dict, return [maybelist], otherwise, return maybelist since it's already iterable.

> let me add a test case to the detection

just added. hope it's formatted correctly 😁

Reply:
sorry i read your comment wrong, i think the snippet you provided above is correct